src/chatd.cpp

```diff
@@ -1975,42 +1985,52 @@ bool Chat::setMessageSeen(Idx idx)
     }
 
     auto wptr = weakHandle();
-    karere::Id id = msg.id();
-    marshallCall([wptr, this, id, idx]()
+    megaHandle mEchoTimer = karere::setTimeout([this, wptr, idx, &msg, &mEchoTimer]()
     {
         if (wptr.deleted())
-            return;
-
-        CHATID_LOG_DEBUG("setMessageSeen: Setting last seen msgid to %s", ID_CSTR(id));
-        sendCommand(Command(OP_SEEN) + mChatId + id);
-        
-        Idx notifyStart;
-        if (mLastSeenIdx == CHATD_IDX_INVALID)
         {
-            notifyStart = lownum()-1;
+          return;
         }
-        else
+        mClient.mSeenTimers.erase(mEchoTimer);
+        mEchoTimer = 0;
```

Comment:
The asignment above is for the pointer, not the value. It doesn't have any effect, so get rid of it.